api/v1beta1/aerospikecluster_validating_webhook.go

```diff
@@ -1060,9 +1063,18 @@ func validateNsConfUpdate(
 				)
 			}
 
-			if singleConf["name"] == oldSingleConf["name"] {
-				found = true
+			// Add oldSingleConf ns device list into deviceList if not already present
+			// Keep track of devices referenced in existing namespaces.
+			oldConfStorage := oldSingleConf["storage-engine"].(map[string]interface{})
+			oldConfDevices := oldConfStorage["devices"].([]string)
+			for _, device := range oldConfDevices {
+				_, exists := deviceList[device]
+				if !exists {
+					deviceList[device] = oldSingleConf["name"].(string)
```

Comment:
.(string) doesn't seem required (name is mandatory)

Reply:
I cannot remove the cast because I get the error:
`cannot use oldSingleConf["name"] (map index expression of type interface{}) as string value in assignment: need type assertion`